core/src/main/java/nl/nn/adapterframework/senders/ShadowSender.java

```diff
@@ -152,14 +152,14 @@ public SenderResult sendMessage(Message message, PipeLineSession session) throws
 				// Collect the results of the (Shadow)Sender and send them to the resultSender.
 				try {
 					Message result = collectResults(executorMap, message, session);
-					resultSender.sendMessageOrThrow(result, session);
+					resultSender.sendMessageOrThrow(result, session).close();
 				} catch (IOException | SAXException e) {
 					log.error("unable to compose result message", e);
-				} catch(TimeoutException | SenderException se) {
-					log.error("failed to send ShadowSender result to ["+resultSender.getName()+"]");
+				} catch (TimeoutException | SenderException se) {
+					log.error("failed to send ShadowSender result to [{}]", resultSender.getName());
```

Comment:
Nou maakt dit voor error level messages niet zo veel uit maar hier ook, suppliers worden pas gecompute op het moment dat ze nodig zijn, mocht hier een methode staan moet hij die altijd eerst aanroepen en geeft hij het resultaat door, ongeacht het loglevel.
```suggestion
					log.error("failed to send ShadowSender result to [{}]", resultSender::getName);
```